mw-webapp/src/component/sidebar/SidebarContent/SidebarContent.tsx

```diff
@@ -42,8 +42,14 @@ export const SidebarContent = (props: SidebarContentProps) => {
    * Handle onClick event
    */
   const onClickHandler = (event: React.MouseEvent<HTMLDivElement>) => {
-    if (event.target instanceof HTMLAnchorElement) {
-      props.onClick();
+    let target: HTMLElement | null = event.target as HTMLElement;
+
+    while (target && target !== event.currentTarget) {
+      if (target instanceof HTMLAnchorElement) {
+        props.onLinkClick();
+        break;
+      }
+      target = target.parentElement;
```

Comment:
> what happened here?)

onClickHandler is called for every click on SidebarContent. The original idea of the code was to close the sidebar only if a click happens on <a> tag (props.onClick). Detection of clicking <a> tag was implemented via condition `if (event.target instanceof HTMLAnchorElement)`. It didn't work correctly because in case of clicking on image inside <a> tag, event.target is not HTMLAnchorElement, event.target is set to deepest element where click occurred (e.g. <img> tag). I fixed this by checking that the clicked element has a parent equal to the <a> tag.

>  Do we have any alternatives?

Yes, I can remove onClickHandler from SidebarContent and set `onClick=()=>setOpen(false)` for every menu link: 

```ts
export const Sidebar = (props: SidebarProps) => {
  const [open, setOpen] = useState(false);
  return (
    <DialogRoot
      open={open}
      onOpenChange={setOpen}
    >
      ....
      <SidebarContent
        ...
      >
        <div className={styles.navSidebarContent}>
          {props.linkList.map((item) => (
            ...
            <Link
              ...
              onClick={() => setOpen(false)}  //HERE!
            >
              {item.icon}
              {item.value}
            </Link>
       ...
    </DialogRoot>
  );
};
```

Reply:
Which variant will be easier to maintain? Original idea was interesting, but I think 2nd variant will be easier a little. What do you think? @Ekaterina1994 @algrinkevich

Reply:
As for me, it's easier to maintain the 2nd variant, because it's pretty straightforward comparing with existed event delegation.

Reply:
ok? lets implement second variant

Reply:
Let's use second variant)

Reply:
Done